### PR TITLE
target level to PID control

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/494.

### CODE REVIEW

File: docs/core/usage.qmd [hunk 1]
```diff
@@ -466,6 +465,7 @@ column         | type     | unit     | restriction
 node_id        | Int      | -        | sorted
 active         | Bool     | -        | (optional, default true)
 listen_node_id | Int      | -        | -
+target_level   | Float64  | $m$      | -
```
Comment (evetion) [inline docs/core/usage.qmd:468]: Is optional

File: docs/schema/PIDControlStatic.schema.json [hunk 2]
```diff
@@ -49,12 +51,17 @@
...
+            "format": "double",
+            "description": "target_level",
+            "type": "number"
         }
     },
     "required": [
         "node_id",
         "listen_node_id",
-        "proportional"
+        "target_level"
```
Comment (evetion) [inline docs/schema/PIDControlStatic.schema.json:64]: Why the change from proportional to target_level? You've used coalesce elsewhere, so that implies it's not always required?

**Comment by evetion** [comment]
@Huite If we move this parameter `target_level` of a Basin to the PID control, I propose to rename to just `target` (or similar generic name), as the PID controller ideally should just know about values (y-intercept and their changes). Do you agree?

File: docs/core/usage.qmd (hunk 1 above)
Comment (SouthEndMusic) [reply docs/core/usage.qmd:468]: How can it be optional? The PID control cannot function without a setpoint. For the Basin it was optional because not all basins have a target level (so before it was required that the PID control node was connected to a basin with a target_level).

File: docs/schema/PIDControlStatic.schema.json (hunk 2 above)
Comment (SouthEndMusic) [reply docs/schema/PIDControlStatic.schema.json:64]: It's not a change from one to the other. Proportional was removed as a required input so that they are all optional. This is handled nicely in the Julia core by skipping the computations for the terms where the coefficient is not defined (NaN). Target level is required as discussed above.

**Comment by gijsber** [comment]
Agree. Use either `target` or `setpoint` as they may be levels or flows

**Comment by SouthEndMusic** [comment]
> @Huite If we move this parameter `target_level` of a Basin to the PID control, I propose to rename to just `target` (or similar generic name), as the PID controller ideally should just know about values (y-intercept and their changes). Do you agree?

@evetion I'll change it to 'target'. 

The implementation of the PID controller is not easily made that generic, variables each have their own algebraic and implementation challenges. For instance if you control flow you run into trouble with the derivative term because flow can change discontinuously over time (e.g. due to `DiscreteControl`). Also, in general certain Jacobian terms are solutions to implicit equations with dependencies on nodes in the vacinity, see https://deltares.github.io/Ribasim/core/equations.html#sec-PID.

**Review by visr — APPROVED** [review]
